firstapp/app/src/androidTest/java/com/example/firstapp/UnitTests.java

```diff
@@ -0,0 +1,61 @@
+package com.example.firstapp;
+import androidx.test.platform.app.InstrumentationRegistry;
+import androidx.test.uiautomator.By;
+import androidx.test.uiautomator.UiDevice;
+import androidx.test.uiautomator.UiObject;
+import androidx.test.uiautomator.UiObject2;
+import androidx.test.uiautomator.UiObjectNotFoundException;
+import androidx.test.uiautomator.UiSelector;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import androidx.test.ext.junit.runners.AndroidJUnit4;
+import androidx.test.uiautomator.Until;
+
+import org.junit.Test;
+
+import java.io.IOException;
+import java.nio.file.attribute.BasicFileAttributes;
+import java.util.prefs.BackingStoreException;
+
+import static org.junit.Assert.*;
+
+/**
+ * Example local unit test, which will execute on the development machine (host).
+ *
+ * @see <a href="http://d.android.com/tools/testing">Testing documentation</a>
+ */
+public class UnitTests {
```

Comment:
Should be a more descriptive name than just UnitTests.

Reply:
should I have different test classes for different functions? My intention with this was to have only one class for all unit tests.

Reply:
Generally, should have a test per class.  For example,

1. UtilityMethods -> UtilityMethodsTest